test/integration/framework/master_utils.go

```diff
@@ -148,28 +150,30 @@ func startMasterOrDie(masterConfig *master.Config) (*master.Master, *httptest.Se
 
 // Returns a basic master config.
 func NewMasterConfig() *master.Config {
-	etcdClient := NewEtcdClient()
-	storageVersions := make(map[string]string)
-
-	etcdStorage := etcdstorage.NewEtcdStorage(etcdClient, testapi.Default.Codec(), etcdtest.PathPrefix(), false)
-	storageVersions[api.GroupName] = testapi.Default.GroupVersion().String()
-	autoscalingEtcdStorage := NewAutoscalingEtcdStorage(etcdClient)
-	storageVersions[autoscaling.GroupName] = testapi.Autoscaling.GroupVersion().String()
-	batchEtcdStorage := NewBatchEtcdStorage(etcdClient)
-	storageVersions[batch.GroupName] = testapi.Batch.GroupVersion().String()
-	expEtcdStorage := NewExtensionsEtcdStorage(etcdClient)
-	storageVersions[extensions.GroupName] = testapi.Extensions.GroupVersion().String()
-
-	storageDestinations := genericapiserver.NewStorageDestinations()
-	storageDestinations.AddAPIGroup(api.GroupName, etcdStorage)
-	storageDestinations.AddAPIGroup(autoscaling.GroupName, autoscalingEtcdStorage)
-	storageDestinations.AddAPIGroup(batch.GroupName, batchEtcdStorage)
-	storageDestinations.AddAPIGroup(extensions.GroupName, expEtcdStorage)
+	etcdConfig := etcdstorage.EtcdConfig{
+		ServerList: []string{"http://127.0.0.1:4001"},
+		Prefix:     etcdtest.PathPrefix(),
+	}
+
+	negotiatedSerializer := serializer.NewNegotiatedSerializerWapper(api.Scheme, testapi.Default.Codec(), "application/json")
```

Comment:
> Why did you make this change adding this wrapper?

The cross group encoding enablement code requires APIs from that interface, so this wrapper exists to plumb the tests through.  I'd actually say the cross group encoding code needs some revisiting since it ends up hard coding serializations formats (@smarterclayton will be unhappy with it later), but I think that refactoring should happen separately.